### PR TITLE
Inbox refactor

### PR DESCRIPTION
**Pull request by sbsmith86**
#### What's this PR do?

Updates the `CampaignInbox` and the `CampaignSingle` components to show posts via the new `Post` component for consistency. 

A `Post` component is responsible for showing the post and allowing the post to be reviewed. I made some updates to the component to give options to allow toggling a modal per post, to show sibling posts, and to show quantity. 

I also [fixed a bug](https://github.com/DoSomething/rogue/commit/f068105d23135b678bd5555ace0364a440f687ba#diff-0335bdd7424f67314da72e781e5e027cR90) where on the Campaign Single view, when a user updated quantity we were always reflecting that new quantity on the first post. 

#### How should this be reviewed?
Commit-by-commit should be safe. Nothing should change functionally. 

#### Any background context you want to provide?
This is the first step in refactoring some of our components. The next step is to figure  out how to share some of the reviewing functionality between components so that we don't have it living in multiple places. 

#### Relevant tickets
Addresses https://www.pivotaltracker.com/story/show/150272139

#### Checklist
- [ ] Documentation added for new features/changed endpoints.
- [ ] Tested on staging.
- [ ] Pinged a PM if this is a larger PR that would benefit from some additional testing love.

### CODE REVIEW

File: app/Http/Controllers/CampaignsController.php [hunk 1]
```diff
@@ -61,19 +61,15 @@ public function showInbox($campaignId)
     {
         $signups = Signup::campaign([$campaignId])->has('pending')->with('pending')->get();
 
+        // Grab the user objects for each signup
+        $ids = $signups->pluck('northstar_id')->toArray();
+        $users = $this->registrar->findAll($ids)->keyBy('id')->map(function ($user, $key) {
+            return $user->toArray();
+        });
+
         // For each pending post, get and include the user
```
Comment (chloealee) [inline app/Http/Controllers/CampaignsController.php]: so much better!

File: resources/assets/components/Signup/index.js [hunk 2]
```diff
@@ -132,7 +132,7 @@ class Signup extends React.Component {
     });
   }
 
-  // Update a signups quanity.
+  // Update a signups quantity.
```
Comment (chloealee) [inline resources/assets/components/Signup/index.js]: :)

File: resources/assets/components/CampaignInbox/index.js [hunk 3]
```diff
@@ -155,8 +158,22 @@ class CampaignInbox extends React.Component {
...
+                signup={signups[post.signup_id]}
+                campaign={campaign}
+                onUpdate={this.updatePost}
+                onTag={this.updateTag}
+                deletePost={this.props.deletePost}
+                showHistory={this.showHistory}
+                showSiblings={true}
+                showQuantity={true}
+                allowHistory={true} />
+            )
```
Comment (katiecrane) [inline resources/assets/components/CampaignInbox/index.js]: This is SO MUCH MORE readable!!!

File: resources/assets/components/InboxItem/index.js [hunk 4]
```diff
@@ -1,138 +0,0 @@
...
-            Source: {post['source']} <br/>
-            Submitted: {post['created_at']} <br/>
-          </p>
-        </div>
-      </div>
-    )
-  }
-}
-
-export default InboxItem;
```
Comment (katiecrane) [inline resources/assets/components/InboxItem/index.js]: 🔥 🔥 🔥

**Pull request merged**